runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8396,7 +8396,15 @@ class INTERPRETER_CLASS
 				if (interfaceClass == iTable->interfaceClass) {
 					receiverClass->lastITable = iTable;
 foundITable:
-					vTableOffset = ((UDATA*)(iTable + 1))[iTableIndex];
+					if (J9_UNEXPECTED(J9_ARE_ANY_BITS_SET(iTableIndex, J9_ITABLE_OFFSET_TAG_BITS))) {
```

Comment:
This fetches J9JNIMethodID* from ` vmindexOffset` and then tries to decode it for an interface.  Looking at the code that initializes the J9JNIMethodID:
https://github.com/eclipse/openj9/blob/a7b7dbe51e2c8801b4f0a22bb98322d7dda76692/runtime/vm/jnicsup.cpp#L2097-L2122

it explicitly states that it doesn't tag the vTableIndex for static interface methods or methods from Object as `J9_JNI_MID_INTERFACE`.

If it's one of those cases, the code below that looks up the method will be wrong:
```
_sendMethod = *(J9Method **)(((UDATA)receiverClass) + vTableOffset);
```

Can you point me at the code that ensures we won't try to run `linkToInterface` for private interface methods or for Object methods looked up on the interface?

With the OpenJ9 implementation, we handled these cases in the Java code that created the MH by returning DirectHandles or VirtualHandles wrapped with appropriate asTypes to ensure we couldn't hit them in the native code.

Reply:
[1] [`DirectMethodHandle` that handles `invokeinterface`](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/1d5907c3350e99332f8bdad8229f7495fd69a4ad/src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java#L457-L483): has `viewAsType`; has `checkReceiver`, which gets called before `linkToInterface`.
[2] [`DirectMethodHandle` check for `invokeinterface` of Object methods](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/1d5907c3350e99332f8bdad8229f7495fd69a4ad/src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java#L62-L72).
[3] In [2], `MethodHandleNatives.resolve` (native code) is used to generate `MemberName`.

Unsure if I have identified the correct code. +@fengxue-IS for assist.

@DanHeidinga can you please provide a link to the corresponding J9 code?

Reply:
Look at 
https://github.com/eclipse/openj9/blob/c757b51154427a436921984a8a4cb316bb5c6cf6/jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java#L839-L855

And at
https://github.com/eclipse/openj9/blob/c757b51154427a436921984a8a4cb316bb5c6cf6/jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java#L1077